### PR TITLE
The Battle for Wesnoth

### PR DESCRIPTION
**Pull request by mattman107**
# Description

Egg for The Battle of Wesnoth server. Needs some specific packages to build/run the server binary so a custom yolk has been created. This egg is written assuming [this PR](https://github.com/pelican-eggs/yolks/pull/237) will be merged.

Since it currently cannot be run until the yolk is merged and built I will keep this a draft.

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [x] Does your egg use a custom docker image?
    * [x] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [x] You verify that the start command applied does not use a shell script
    * [ ] If some script is needed then it is part of a current yolk or a PR to add one
6. [x] The egg was exported from the panel

### CODE REVIEW

File: thebattleforwesnoth/egg-battle-for-wesnoth.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    "exported_at": "2024-04-20T10:04:36-04:00",
+    "name": "The Battle for Wesnoth",
+    "author": "mattman107@github.com",
+    "description": "Egg for hosting a Battle for Wesnoth Server.",
+    "features": null,
+    "docker_images": {
+        "ghcr.io\/parkervcp\/games:thebattleforwesnoth": "ghcr.io\/parkervcp\/games:thebattleforwesnoth"
+    },
+    "file_denylist": [],
+    "startup": "echo -e \"passwd=\\\"{{PASSWD}}\\\"\" > wesnothd.cfg; echo -e \"motd=\\\"{{MOTD}}\\\"\" >> wesnothd.cfg; .\/battleForWesnoth\/build\/wesnothd -p {{SERVER_PORT}} -c \/home\/container\/wesnothd.cfg",
```
Comment (QuintenQVD0) [inline thebattleforwesnoth/egg-battle-for-wesnoth.json:16]: this egg things hsould be done with the parser and no directly echo to the file.

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]

File: thebattleforwesnoth/egg-battle-for-wesnoth.json (hunk 1 above)
Comment (mattman107) [reply thebattleforwesnoth/egg-battle-for-wesnoth.json:16]: As far as I understand from the [documentation](https://pterodactyl.io/community/config/eggs/creating_a_custom_egg.html#configuration-files) and the [source code](https://github.com/pterodactyl/wings/blob/develop/parser/parser.go#L25-L30),  '.cfg' isn't a valid file parsing option. Do one of the other options also work for '.cfg' files?

Comment (QuintenQVD0) [reply thebattleforwesnoth/egg-battle-for-wesnoth.json:16]: use the file one, it is made to replace whole lines in files.

Comment (mattman107) [reply thebattleforwesnoth/egg-battle-for-wesnoth.json]: Updated to use the file config parser.

**Comment by QuintenQVD0** [comment]
@mattman107 Please update your fork branch as it is missing to many commits to review / work on this

**Review by QuintenQVD0 — APPROVED** [review]
LGTM! Thank you.

**Pull request merged**